File: Data Structures & Algorithms/max-water-container/submission-0.java
class Solution {
    public int maxArea(int[] heights) {
        int max = 0;
        for (int i = 0; i < heights.length; i++) {
            int j = heights.length - 1;

            while (j > i) {
                int volume = Math.min(heights[i], heights[j]) * (j - i);
                if (volume > max) {
                    max = volume;
                }
                j--;
            }
        }
        return max;
    }
}

File: Data Structures & Algorithms/pacific-atlantic-water-flow/submission-0.py
class Solution:
    def pacificAtlantic(self, heights: List[List[int]]) -> List[List[int]]:
        ROWS, COLS = len(heights), len(heights[0])
        pac, atl = set(), set()

        def dfs(r, c, prev, visit):
            if (r,c) in visit or min(r, c) < 0 or r == ROWS or c == COLS or heights[r][c] < prev:
                return
            visit.add((r, c))
            dfs(r + 1, c, heights[r][c], visit)
            dfs(r - 1, c, heights[r][c], visit)
            dfs(r, c + 1, heights[r][c], visit)
            dfs(r, c - 1, heights[r][c], visit)

        for c in range(COLS):
            dfs(0, c, 0, pac)
            dfs(ROWS - 1, c, 0, atl)

        for r in range(ROWS):
            dfs(r, 0, 0, pac)
            dfs(r, COLS - 1, 0, atl)
        
        res = []
        for r in range(ROWS):
            for c in range(COLS):
                if (r, c) in pac and (r, c) in atl:
                    res.append([r, c])

        return res
        
            